src/emulator/net/src/net.cpp

```diff
@@ -0,0 +1,160 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include <net/functions.h>
+
+#include <net/state.h>
+
+#include <cassert>
+#include <cstring>
+
+static void convertSceSockaddrToPosix(struct SceNetSockaddr *src, struct sockaddr *dst){
+    dst->sa_family = src->sa_family;
+    memcpy(&dst->sa_data, &src->sa_data, 14);
+}
+
+static void convertPosixSockaddrToSce(struct sockaddr *src, struct SceNetSockaddr *dst){
+    dst->sa_family = src->sa_family;
```

Comment:
Can src be const? We could skip some casts later if so.

Reply:
Making this const will just add even more casts since that function is called only for converting a temp non-const (since it came from a previous call of convertSceSockaddrToPosix) struct to a Sce struct as result.